### PR TITLE
[WIP] Issue 16 edit account settings

### PR DESCRIPTION
**Pull request by hechi**
not done yet.
@ChristophWurst please have a look views/accountsettings.js the trigger for onAccountSubmit is not triggered :(

### CODE REVIEW

File: lib/controller/accountscontroller.php [hunk 1]
```diff
@@ -150,11 +151,20 @@ public function show($accountId) {
 
 	/**
 	 * @NoAdminRequired
+	 *
+	 * @param int $accountId
+	 * @return JSONResponse
 	 */
-	public function update() {
+	public function edit($accountId) {
```
Comment (ChristophWurst) [inline lib/controller/accountscontroller.php]: I think the method should be named `update` since this is a resource controller.

Comment (ChristophWurst) [inline lib/controller/accountscontroller.php]: see https://github.com/nextcloud/mail/blob/ca1f55b7b05e86484225195561f9e8fe67c85792/appinfo/routes.php#L63

File: lib/controller/accountscontroller.php [hunk 2]
```diff
@@ -150,11 +151,20 @@ public function show($accountId) {
...
 		$response = new Response();
 		$response->setStatus(Http::STATUS_NOT_IMPLEMENTED);
```
Comment (ChristophWurst) [inline lib/controller/accountscontroller.php]: you might want to remove this line ;-)

File: js/views/accountsettings.js [hunk 3]
```diff
@@ -24,37 +24,77 @@ define(function(require) {
...
+			'smtpUser' : 'input[name="smtp-user"]',
+			'smtpPort' : 'input[name="smtp-port"]',
+			'smtpPassword' : 'input[name="smtp-password"]',
+			'submitAccountButton': 'input[name="submit-account-settings"]',
 		},
 		events: {
-			'click @ui.submitButton': 'onSubmit',
-			'submit @ui.form': 'onSubmit'
+			'click @ui.submitAliasButton': 'onAliasSubmit',
+			'click @ui.submitAccountSettingsButton': 'onAccountSubmit'
```
Comment (ChristophWurst) [inline js/views/accountsettings.js]: try `@ui.submitAccountButton` 🙈

**Comment by ChristophWurst** [comment]
FYI: we prefer rebasing over merging to update feature branches 😉

**Comment by hechi** [comment]
@ChristophWurst thanks, i will do rebase :)
do you have an idea how i can load the old selection for the none/tls/ssl modes in the template?

**Comment by hechi** [comment]
hmm... i am not sure if this rebase was correct ^^

**Comment by ChristophWurst** [comment]
whoops, I'm pretty sure it wasn't 🙈 🙉 🙊

**Comment by hechi** [comment]
@ChristophWurst now everything should work expect loading and selecting the right sslMode :( can you help me? I have no idea how i can use the LayoutView to preselect the option.

**Comment by hechi** [comment]
i close this PR and open a new one for the real PR, because screwed up the PR with rebase.

**Pull request closed without merging**